crates/services/relayer/src/ports.rs

```diff
@@ -1,27 +1,133 @@
 //! Ports used by the relayer to access the outside world
 
 use async_trait::async_trait;
-use fuel_core_storage::Result as StorageResult;
+use fuel_core_storage::{
+    not_found,
+    tables::Messages,
+    transactional::Transactional,
+    Error as StorageError,
+    Mappable,
+    Result as StorageResult,
+    StorageAsMut,
+    StorageAsRef,
+    StorageMutate,
+};
 use fuel_core_types::{
     blockchain::primitives::DaBlockHeight,
-    entities::message::{
-        CheckedMessage,
-        Message,
-    },
+    entities::message::CheckedMessage,
 };
 
+#[cfg(test)]
+mod tests;
+
 /// Manages state related to supported external chains.
 #[async_trait]
 pub trait RelayerDb: Send + Sync {
-    /// Add bridge message to database. Messages are not revertible.
-    fn insert_message(
-        &mut self,
-        message: &CheckedMessage,
-    ) -> StorageResult<Option<Message>>;
+    /// Add bridge messages to database. Messages are not revertible.
+    /// Must set the maximum da height for these messages in
+    /// the same write transaction.
+    /// Must only set a new da height if it is greater than the current.
+    fn insert_messages(&mut self, messages: &[CheckedMessage]) -> StorageResult<()>;
```

Comment:
We don't need to require the calculation of max height by the database based on messages. We need to pass the height of the finalization from the relayer.

Like "Hey, database, please save the final state of the synchronization with those messages at X height".